tensorflow_addons/optimizers/python/lazy_adam_optimizer.py

```diff
@@ -54,30 +52,32 @@ def _resource_apply_sparse(self, grad, var, indices):
         lr_t = self._decayed_lr(var_dtype)
         beta_1_t = self._get_hyper('beta_1', var_dtype)
         beta_2_t = self._get_hyper('beta_2', var_dtype)
-        local_step = math_ops.cast(self.iterations + 1, var_dtype)
-        beta_1_power = math_ops.pow(beta_1_t, local_step)
-        beta_2_power = math_ops.pow(beta_2_t, local_step)
+        local_step = tf.cast(self.iterations + 1, var_dtype)
+        beta_1_power = tf.math.pow(beta_1_t, local_step)
+        beta_2_power = tf.math.pow(beta_2_t, local_step)
         epsilon_t = self._get_hyper('epsilon', var_dtype)
-        lr = (lr_t * math_ops.sqrt(1 - beta_2_power) / (1 - beta_1_power))
+        lr = (lr_t * tf.math.sqrt(1 - beta_2_power) / (1 - beta_1_power))
 
         # \\(m := beta1 * m + (1 - beta1) * g_t\\)
         m = self.get_slot(var, "m")
-        m_t_slice = beta_1_t * array_ops.gather(
+        m_t_slice = beta_1_t * tf.gather(
             m, indices) + (1 - beta_1_t) * grad
         m_update_op = resource_variable_ops.resource_scatter_update(
             m.handle, indices, m_t_slice)
 
         # \\(v := beta2 * v + (1 - beta2) * (g_t * g_t)\\)
         v = self.get_slot(var, "v")
-        v_t_slice = (beta_2_t * array_ops.gather(v, indices) +
-                     (1 - beta_2_t) * math_ops.square(grad))
+        v_t_slice = (beta_2_t * tf.gather(v, indices) +
+                     (1 - beta_2_t) * tf.math.square(grad))
+
         v_update_op = resource_variable_ops.resource_scatter_update(
             v.handle, indices, v_t_slice)
 
         # \\(variable -= learning_rate * m_t / (epsilon_t + sqrt(v_t))\\)
-        var_slice = lr * m_t_slice / (math_ops.sqrt(v_t_slice) + epsilon_t)
+        var_slice = lr * m_t_slice / (tf.math.sqrt(v_t_slice) + epsilon_t)
+
+        # var_update_op = tf.raw_ops.ResourceScatterSub(
```

Comment:
Also unable to get this `raw_ops` to replace the below resource_variable api call

Reply:
Hm, the raw ops should be there-- what error are you seeing?

Reply:
Without digging too much into this it looks like some wrapper is expecting no params though 3 are being passed from `resource_scatter_sub`
```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 1101, in decorated
    return f(self, *args, **kwargs)
  File "/root/.cache/bazel/_bazel_root/c27e9809996ce9a9c0ed8dd79ef0897b/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/optimizers/lazy_adam_optimizer_test.runfiles/__main__/tensorflow_addons/optimizers/python/lazy_adam_optimizer_test.py", line 148, in testSparseRepeatedIndices
    [(grad_repeated_index, repeated_index_update_var)])
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 406, in apply_gradients
    self._distributed_apply, args=(grads_and_vars,), kwargs={"name": name})
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 1375, in merge_call
    return self._merge_call(merge_fn, args, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 1382, in _merge_call
    return merge_fn(self._strategy, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 440, in _distributed_apply
    var, apply_grad_to_update_var, args=(grad,), group=False))
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 1175, in update
    return self._update(var, fn, args, kwargs, group)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 1546, in _update
    return self._update_non_slot(var, fn, (var,) + tuple(args), kwargs, group)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 1552, in _update_non_slot
    result = fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 424, in apply_grad_to_update_var
    grad.values, var, grad.indices)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 799, in _resource_apply_sparse_duplicate_indices
    return self._resource_apply_sparse(summed_grad, handle, unique_indices)
  File "/root/.cache/bazel/_bazel_root/c27e9809996ce9a9c0ed8dd79ef0897b/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/optimizers/lazy_adam_optimizer_test.runfiles/__main__/tensorflow_addons/optimizers/python/lazy_adam_optimizer.py", line 79, in _resource_apply_sparse
    var_update_op = tf.raw_ops.ResourceScatterSub(var.handle, indices, var_slice)
TypeError: wrapper() takes exactly 0 arguments (3 given)
```

Reply:
For raw ops, all args must be kwargs. (That error message could be better-- I will file a bug for that.)

Reply:
Ty, pushed a fix.